src/core/DataFile.cpp

```diff
@@ -326,6 +336,139 @@ bool DataFile::writeFile( const QString& filename )
 
 
 
+// TODO: Remove qWarnings
+bool DataFile::writeBundle(const QString& filename)
+{
+	qWarning() << "Write Bundle DEBUG:";
+
+	const QString fullName = nameWithExtension(filename);
+	const QString fullNameTemp = fullName + ".new";
+	const QString resourcesDirName = QFileInfo(fullName).path() + "/resources";
+
+	qWarning() << "Destination File: " << fullName;
+	qWarning() << "Resources Folder: " << resourcesDirName;
+
+	QFile outfile(fullNameTemp);
+
+	if (!outfile.open(QIODevice::WriteOnly | QIODevice::Truncate))
+	{
+		if (gui)
+		{
+			QMessageBox::critical(NULL,
+				SongEditor::tr("Could not write file"),
+				SongEditor::tr("Could not open %1 for writing. You probably are not permitted to "
+								"write to this file. Please make sure you have write-access to "
+								"the file and try again.").arg(fullName));
+		}
+		qWarning() << "ERROR: Failed to create bundle file!";
+		return false;
+	}
+
+	// Creates resources folder
+	if (QDir(resourcesDirName).exists())
+	{
+		qWarning() << "Warning: Resources dir already exists. Proceeding anyways.";
+	}
+	else
+	{
+		if(!QDir().mkdir(resourcesDirName))
+		{
+			qWarning() << "ERROR: Failed to create resources dir!";
+			return false;
+		}
+	}
```

Comment:
Not sure if this in intention or not, but if the resources directory exists that's okay right? If so, then we can just check for its non-existence.
```suggestion
	if (!QDir(resourcesDirName).exists())
	{
		qWarning() << "resources dir does not exist, creating";
		if (!QDir().mkdir(resourcesDirName))
		{
			qWarning() << "ERROR: Failed to create resources dir!";
			return false;
		}
	}
```

Reply:
For now I made it Ok, but we can decide whether LMMS should abort creating a bundle if the resources folder already exists. Think it would be safer to abort, what do you think?

Reply:
This would allow only a single bundle to be made at a time, which requires manually moving/removing folders outside of lmms functionality. I don't particularly like that idea. I'd rather we figure out a way to handle it within lmms itself. What if we detect the folder presence and return a list of contents and ask the user what they want to do (abort/clear folder)?

Reply:
You would still be able to create multiple bundles, just not on the same folder (which kind of makes sense, since the bundle right now is a folder tree basically consisting of the project file and the resources directory). Allowing bundles to be created on the same folder can become a little bit of a mess, since the resources of both projects will get mixed in the same folder. What I think could be improved though is to always create a folder with the project name, inside which there will be the project file and resources dir, which now will be guaranteed to be empty since the folder was just created.

Reply:
What happens if they try to save the same project again?

Reply:
When I originally thought of this feature I think I had in mind that "Save bundle" would always behave as "Save as". But maybe saving over an existing bundle would be desirable? It still could get messy to allow the same folder to be reused though, we could ask for user confirmation and just clear the folder before copying the new resources. Guess I didn't think as far as that :p

Reply:
> Guess I didn't think as far as that :p

That's what I'm here for. ;)

Reply:
I was thinking about asking for a bundle name instead of a project file name. The folder would be created and inside it there would be the project file and the resources folder (sort of like how Ardour saves its project files). If the user chooses a name and there's a folder with that name already the operation would be aborted and there would be a warning saying a new name has to be chosen (I thought allowing overwriting could get very dangerous if the user accidentally opts to overwrite the wrong folder, since everything inside it would have to be deleted first). What do you think of that behavior?

Reply:
That should work!